16/umbraco-cms/customizing/extending-overview/extension-types/entity-actions.md

```diff
@@ -1,170 +1,189 @@
-## Registering an Entity Action <a href="#registering-an-entity-action" id="registering-an-entity-action"></a>
+### Declare the Entity Action
 
+{% code title="entity-action/manifest.ts" %}
 ```typescript
 import { extensionRegistry } from '@umbraco-cms/extension-registry';
-import { MyEntityAction } from './entity-action';
 
 const manifest = {
-	type: 'entityAction',
-	alias: 'My.EntityAction',
-	name: 'My Entity Action',
-	weight: 10,
-	api: MyEntityAction,
-	forEntityTypes: ['my-entity'],
-	meta: {
-		icon: 'icon-add',
-		label: 'My Entity Action',
-		repositoryAlias: 'My.Repository',
-	},
+    type: 'entityAction',
+    alias: 'My.EntityAction',
+    name: 'My Entity Action',
+    weight: 10,
+    api: () => import('./my-entity-action.js'),
+    forEntityTypes: ['my-entity'],
+    meta: {
+        icon: 'icon-add',
+        label: 'My Entity Action',
+    },
 };
 
 extensionRegistry.register(manifest);
 ```
+{% endcode %}
 
-**Default Element**
+## The Entity Action Class <a href="#the-entity-action-class" id="the-entity-action-class"></a>
 
-```typescript
-interface UmbEntityActionElement {}
-```
+Umbraco provides a few generic actions that can be used across silos, such as copy, move, trash, etc. Umbraco may include additional generic actions in the future.
+
+Entity Action extensions will need to supply a class to the extension definition using the `api` property in the manifest file. This class will be instantiated as part of the action and will be passed a reference to the entity that invoked it.
+
+The entity action class will provide one of the following methods:
+* `getHref` - returns a URL that will be used for navigation
+* `execute` - programs custom imperative behaviors that can work with contexts and service APIs
 
-### The Entity Action Class <a href="#the-entity-action-class" id="the-entity-action-class"></a>
+If both methods are provided in the entity action class, the `getHref` method will be preferred.
 
-As part of the Extension Manifest you can attach a class that will be instantiated as part of the action. It will have access to the host element, a repository with the given alias and the unique (key etc) of the entity.
+When the action is completed, an event on the host element will be dispatched to notify any surrounding elements.
 
-The class either provides a getHref method, or an execute method. If the getHref method is provided, the action will use the link. Otherwise the `execute` method will be used. When the action is clicked the `execute` method on the api class will be run. When the action is completed, an event on the host element will be dispatched to notify any surrounding elements.
+### The `getHref()` Method <a href="#get-href-method" id="get-href-method"></a>
 
-Example of providing a `getHref` method:
+Entity action extensions are provided `this.args` by the `UmbEntityActionBase` superclass. The `this.args` contains a property, `unique` that allows developers to identity which element the user selected.
```

Comment:
I'm not entirely sure I understand the first sentence here. Is `this.args` provided by `UmbEntityActionBase`? If that's the case, it needs to be a little clearer.

```suggestion
Entity action extensions are provided `this.args` by the `UmbEntityActionBase` superclass. The `this.args` contains a `unique` property that allows developers to identify which element the user selected.
```